src/engraving/draw/color.cpp

```diff
@@ -0,0 +1,489 @@
+    : Color(QColor(color))
+{
+}
+
+#endif
+
+mu::draw::Color& mu::draw::Color::operator=(const Color& other)
+{
+    m_rgb = other.m_rgb;
+    return *this;
+}
+
+mu::draw::Color& mu::draw::Color::operator=(mu::draw::GlobalColor color)
+{
+    *this = Color(color);
+    return *this;
+}
+
+#ifndef NO_QT_SUPPORT
+mu::draw::Color& mu::draw::Color::operator=(const QColor& other)
+{
+    *this = fromQColor(other);
+    return *this;
+}
+
+mu::draw::Color& mu::draw::Color::operator=(Qt::GlobalColor color)
+{
+    *this = Color::fromQColor(color);
+    return *this;
+}
+
+#endif
+
+bool mu::draw::Color::operator==(const Color& other) const noexcept
+{
+    return m_rgb == other.m_rgb;
+}
+
+bool mu::draw::Color::operator!=(const Color& other) const noexcept
+{
+    return m_rgb != other.m_rgb;
+}
+
+bool mu::draw::Color::operator==(const GlobalColor& color) const noexcept
+{
+    return *this == Color(color);
+}
+
+bool mu::draw::Color::operator!=(const GlobalColor& color) const noexcept
+{
+    return *this != Color(color);
+}
+
+bool mu::draw::Color::isValid() const noexcept
+{
+    return m_isValid;
+}
+
+int mu::draw::Color::red() const noexcept
+{
+    return getRed(m_rgb);
+}
+
+int mu::draw::Color::green() const noexcept
+{
+    return getGreen(m_rgb);
+}
+
+int mu::draw::Color::blue() const noexcept
+{
+    return getBlue(m_rgb);
+}
+
+int mu::draw::Color::alpha() const noexcept
+{
+    return getAlpha(m_rgb);
+}
+
+std::string mu::draw::Color::name() const noexcept
+{
+    return m_name;
+}
+
+#ifndef NO_QT_SUPPORT
+mu::draw::Color mu::draw::Color::fromQColor(const QColor& color)
+{
+    Color c;
+    c.m_rgb = color.rgba();
+    return c;
+}
+
+QColor mu::draw::Color::toQColor() const
+{
+    return QColor(m_rgb);
+}
+
+#endif
+
+void mu::draw::Color::setNamedColor(const std::string& color) noexcept
+{
+    if (color.size() == 0) {
+        m_isValid = false;
+        return;
+    }
+
+    Rgb rgb;
+    if (color[0] == '#') {
+        if (get_hex_rgb(color.data(), color.size(), &rgb)) {
+            setRgb(rgb);
+        } else {
+            m_isValid = false;
+        }
+
+        return;
+    }
+
+    if (get_named_rgb(color.data(), color.size(), &rgb)) {
+        setRgb(rgb);
+    } else {
+        m_isValid = false;
+    }
+}
+
+void mu::draw::Color::setNamedColor(const char* color) noexcept
+{
+    setNamedColor(std::string(color));
+}
+
+void mu::draw::Color::setRed(int value) noexcept
+{
+    setRgb(value, green(), blue(), alpha());
+}
+
+void mu::draw::Color::setGreen(int value) noexcept
+{
+    setRgb(red(), value, blue(), alpha());
+}
+
+void mu::draw::Color::setBlue(int value) noexcept
+{
+    setRgb(red(), green(), value, alpha());
+}
+
+void mu::draw::Color::setAlpha(int value) noexcept
+{
+    setRgb(red(), green(), blue(), value);
+}
+
+void mu::draw::Color::setRgb(int r, int g, int b, int a)
+{
+    if (!isRgbaValid(r, g, b, a)) {
+        m_isValid = false;
+        return;
+    }
+
+    m_rgb = rgba(r, g, b, a);
+}
```

Comment:
There is a method
```
std::string mu::draw::Color::name() const noexcept
{
    return m_name;
}
```
But we can set individual color components
The problem is that `name` will return either the old or empty value.